presto-testing-server-launcher/src/main/java/com/facebook/presto/server/testing/TestingPrestoServerLauncher.java

```diff
@@ -0,0 +1,103 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.testing;
+
+import com.facebook.presto.server.testing.TestingPrestoServerLauncherOptions.Catalog;
+import com.facebook.presto.spi.Plugin;
+import io.airlift.command.Command;
+import io.airlift.command.Help;
+import io.airlift.command.HelpOption;
+import io.airlift.command.model.CommandMetadata;
+
+import javax.inject.Inject;
+
+import static io.airlift.command.SingleCommand.singleCommand;
+import static java.lang.Runtime.getRuntime;
+
+@Command(name = "testing_presto_server", description = "Testing Presto Server Launcher")
+public class TestingPrestoServerLauncher
+{
+    @Inject CommandMetadata commandMetadata;
+
+    @Inject
+    public HelpOption helpOption;
+
+    @Inject
+    TestingPrestoServerLauncherOptions options = new TestingPrestoServerLauncherOptions();
+
+    private static void registerServerCloseShutdownHook(final TestingPrestoServer server)
+    {
+        getRuntime().addShutdownHook(new Thread(() -> server.close()));
+    }
+
+    private static void waitForInterruption()
+    {
+        try {
+            Thread.currentThread().join();
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public void run()
+            throws Exception
+    {
+        TestingPrestoServer server = new TestingPrestoServer();
+        registerServerCloseShutdownHook(server);
+        for (String pluginClass : options.getPluginClassNames()) {
+            Plugin plugin = (Plugin) Class.forName(pluginClass).newInstance();
+            server.installPlugin(plugin);
+        }
+
+        for (Catalog catalog : options.getCatalogs()) {
+            server.createCatalog(catalog.getCatalogName(), catalog.getConnectorName());
+        }
+
+        System.out.println(server.getAddress());
+        waitForInterruption();
+    }
+
+    public static void main(String[] args)
+            throws Exception
+    {
+        try {
+            TestingPrestoServerLauncher launcher = singleCommand(TestingPrestoServerLauncher.class).parse(args);
+            if (launcher.helpOption.showHelpIfRequested()) {
+                return;
+            }
+            try {
+                launcher.validateOptions();
+            }
+            catch (IllegalStateException e) {
+                System.out.println("ERROR: " + e.getMessage());
+                System.out.println();
+                Help.help(launcher.commandMetadata);
+                return;
+            }
+            launcher.run();
+        }
+        catch (Exception e) {
+            e.printStackTrace(System.err);
+        }
+        finally {
+            System.exit(0);
```

Comment:
Removed. With finally instead ShutDownHook all the threads are finishing properly.